### PR TITLE
refactor!: Replace `Surface` `FreeVector` input with position and direction

### PR DESCRIPTION
**Pull request by andiwand**
We do not depend on free parameters for `Surface` so this should not be part of the interface. I consistently replace this by position and direction vectors here.

blocked by
- https://github.com/acts-project/acts/pull/2932
- https://github.com/acts-project/acts/pull/2811

### CODE REVIEW

File: Core/src/Surfaces/DiscSurface.cpp [hunk 1]
```diff
@@ -209,21 +210,18 @@ Acts::Vector2 Acts::DiscSurface::localCartesianToPolar(
...
   RotationMatrix3 rframeT =
       referenceFrame(gctx, position, direction).transpose();
+
   // calculate the transformation to local coordinates
-  const Vector3 posLoc = transform(gctx).inverse() * position;
-  const double lr = perp(posLoc);
-  const double lphi = phi(posLoc);
+  const Vector3 pos_loc = transform(gctx).inverse() * position;
+  const double lr = perp(pos_loc);
+  const double lphi = phi(pos_loc);
```
Comment (AJPfleger) [inline Core/src/Surfaces/DiscSurface.cpp:224]:
```suggestion
  const Vector3 posLoc = transform(gctx).inverse() * position;
  const double lr = perp(posLoc);
  const double lphi = phi(posLoc);
```
is there a reason for nonCamelCase?

File: Tests/UnitTests/Core/EventData/CorrectedTransformFreeToBoundTests.cpp [hunk 2]
```diff
@@ -91,7 +91,7 @@ BOOST_AUTO_TEST_CASE(CorrectedFreeToBoundTrackParameters) {
 
   // the jacobian from local to global at the starting position
   BoundToFreeMatrix boundToFreeJac =
-      eSurface->boundToFreeJacobian(geoCtx, eFreeParams);
+      eSurface->boundToFreeJacobian(geoCtx, tpos, dir);
```
Comment (AJPfleger) [inline Tests/UnitTests/Core/EventData/CorrectedTransformFreeToBoundTests.cpp:94]: do we need the free parameter vector construction at all in this test?

```
  // construct the free parameters vector
  FreeVector eFreeParams = FreeVector::Zero();
  eFreeParams.segment<3>(eFreePos0) = tpos;
  eFreeParams[eFreeTime] = t;
  eFreeParams.segment<3>(eFreeDir0) = dir;
eFreeParams[eFreeQOverP] = qOverP;
```

Comment (andiwand) [reply Tests/UnitTests/Core/EventData/CorrectedTransformFreeToBoundTests.cpp:94]: they are still in use a bit further down
```c++
  auto correctedRes = transformer(eFreeParams, eFreeCov, *eSurface, geoCtx);
```

**Comment by andiwand** [comment]
triggered athena - this will likely need some updates lets see

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [dcf58a56269ed65ea454113fea404260fab972f8]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37644619) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7156600)!

**Review by AJPfleger — APPROVED** [review]
thanks for patching athena. lgtm

**Pull request merged**